src/Api/Github.php

```diff
@@ -26,30 +26,32 @@ class Github extends AbstractApi
      *
      * @param string $origin The repository remote origin.
      *
-     * @param string $user The API username.
+     * @param string $hostname The hostname of GitHub service.
      *
-     * @param string $token The API secret token.
+     * @param string $user   The API username.
      *
+     * @param string $token  The API secret token.
      */
-    public function __construct($origin, $user, $token)
+    public function __construct($origin, $hostname, $user, $token)
     {
+        // @see https://developer.github.com/v3/enterprise
+        if (strpos($hostname, 'github.com') === false) {
+            $hostname .= '/api/v3';
+        }
+
         // set the HTTP object
-        $this->setHttp("https://{$user}:{$token}@api.github.com");
+        $this->setHttp("https://{$user}:{$token}@{$hostname}");
+        
+        $this->setRepoNameFromOrigin($origin);
+    }
 
+    private function setRepoNameFromOrigin($origin)
+    {
         // start by presuming HTTPS
         $repoName = parse_url($origin, PHP_URL_PATH);
 
-        // check for SSH
-        $ssh = 'git@github.com:';
-        $len = strlen($ssh);
-        if (substr($origin, 0, $len) == $ssh) {
-            $repoName = substr($origin, $len);
-        }
-
-        // strip .git from the end
-        if (substr($repoName, -4) == '.git') {
-            $repoName = substr($repoName, 0, -4);
-        }
+        $repoName = explode(':', $repoName)[1];
+        $repoName = rtrim($repoName, '.git');
```

Comment:
With [`rtrim()`](http://php.net/rtrim), the second param is a character mask, not a literal string, so this is not quite the same thing as removing a literal `.git` from the end. It probably *works* but it doesn't convey the right intent and might have unexpected side-effects.